Dosdebug 102

andrewbird opened the pull request:
Some more dosdebug tidy-ups

andrewbird committed:
dosdebug: Fixup inconsistent formatting [skip ci]

Note: no functional changes

andrewbird committed:
dosdebug: Ensure DOS load parameter block is correct size

andrewbird committed:
dosdebug: Avoid duplicating far pointer structure

stsp commented:
> When the dpmimode changed between setting and hitting a breakpoint
> an incorrect mode was restored when the breakpoint was hit.

Well, that's a poor description,
as you actually do a serious
functionality change.
For the mere fix, just this
`dpmimode = mhpdbgc.brktab[i1].is_dpmi;`
would be enough.

stsp commented:
So I suggest splitting that.

andrewbird committed:
dosdebug: Don't update the dpmimode when the breakpoint is hit

When the dpmimode changed between setting and hitting a breakpoint
an incorrect mode was restored when the breakpoint was hit.

andrewbird commented:
Sure, I struggled to come up with a description for the whole change. Now only the bit I really care about is changed.

stsp commented on src/plugin/debugger/mhpdbgc.c:
```diff
@@ -2485,7 +2485,6 @@ static int bpchk(unsigned int a1)
 
   for (i1 = 0; i1 < MAXBP; i1++) {
     if (mhpdbgc.brktab[i1].is_valid && mhpdbgc.brktab[i1].brkaddr == a1) {
```
The condition probably had to
be changed to
`&& dpmimode == mhpdbgc.brktab[i1].is_dpmi`
to make any sense of this `is_dpmi`
after your removal. But not something
to really care about.

stsp commented:
Thanks!
I don't think this can fail CI,
so lets apply & sleep. :)
